/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
Exception in thread Exception in thread Thread-1Thread-1:
:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
        pubsub.get_message(ignore_subscribe_messages=True,response = self.parse_response(block=False, timeout=timeout)

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 321, in can_read
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    return self._buffer and self._buffer.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 230, in can_read
    self._read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 201, in _read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 321, in can_read
    return self._buffer and self._buffer.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 230, in can_read
    self._read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 201, in _read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
15:38:31 Traceback (most recent call last):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 1013, in perform_job
    rv = job.perform()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/job.py", line 709, in perform
    self._result = self._execute()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/job.py", line 732, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/background_jobs.py", line 104, in execute_job
    method(**kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 997, in create_contact
    contact.insert(ignore_permissions=True, ignore_links=ignore_links, ignore_mandatory=ignore_mandatory)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/model/document.py", line 232, in insert
    self._validate_links()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/model/document.py", line 828, in _validate_links
    frappe.throw(_("Could not find {0}").format(msg),
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 438, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red', is_minimizable=is_minimizable, wide=wide, as_list=as_list)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 417, in msgprint
    _raise_exception()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 371, in _raise_exception
    raise raise_exception(msg)
frappe.exceptions.LinkValidationError: Could not find User Id: a@a.com

15:38:32 Traceback (most recent call last):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 1013, in perform_job
    rv = job.perform()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/job.py", line 709, in perform
    self._result = self._execute()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/job.py", line 732, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/background_jobs.py", line 104, in execute_job
    method(**kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 997, in create_contact
    contact.insert(ignore_permissions=True, ignore_links=ignore_links, ignore_mandatory=ignore_mandatory)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/model/document.py", line 232, in insert
    self._validate_links()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/model/document.py", line 828, in _validate_links
    frappe.throw(_("Could not find {0}").format(msg),
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 438, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red', is_minimizable=is_minimizable, wide=wide, as_list=as_list)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 417, in msgprint
    _raise_exception()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/__init__.py", line 371, in _raise_exception
    raise raise_exception(msg)
frappe.exceptions.LinkValidationError: Could not find User Id: a@a.com

/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
Exception in thread Exception in thread Thread-1:
Thread-1:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
            self.run()self.run()

self.run()  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    if not block and not conn.can_read(timeout=timeout):    
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
    redis.exceptionsreturn self._parser.can_read(timeout).
ConnectionError:   File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
Connection closed by server.
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
Traceback (most recent call last):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 559, in connect
Traceback (most recent call last):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 559, in connect
    sock = self._connect()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 615, in _connect
        sock = self._connect()raise err

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 615, in _connect
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 603, in _connect
        raise errsock.connect(socket_address)

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 603, in _connect
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    sock.connect(socket_address)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 104, in <module>
    Exception in thread Thread-1:
Traceback (most recent call last):
      File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
main()return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 19, in main
    exec(code, run_globals)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 104, in <module>
    click.Group(commands=commands)(prog_name='bench')
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    main()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 19, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 782, in main
    return self.main(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 782, in main
Traceback (most recent call last):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 427, in read_from_socket
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    rv = self.invoke(ctx)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    bufflen = recv_into(self._sock, self._buffer)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/_compat.py", line 75, in recv_into
    return sock.recv_into(*args, **kwargs)    
self.run()
ConnectionResetError  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 104, in <module>
    main()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 19, in main
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    click.Group(commands=commands)(prog_name='bench')
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 829, in __call__
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    return callback(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/commands/scheduler.py", line 178, in start_worker
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    start_worker(queue, quiet = quiet)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/background_jobs.py", line 155, in start_worker
    Worker(queues, name=get_worker_name(queue)).work(logging_level = logging_level)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 626, in work
    self.register_death()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 326, in register_death
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    p.execute()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 4012, in execute
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    conn = self.connection_pool.get_connection('MULTI',
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 1192, in get_connection
    return self._parser.can_read(timeout)
    response = self.parse_response(block=False, timeout=timeout)  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    connection.connect()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost:11000. Connection refused.
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    rv = self.invoke(ctx)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return self.main(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 782, in main
        response = self.parse_response(block=False, timeout=timeout)return _process_result(sub_ctx.command.invoke(sub_ctx))

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    rv = self.invoke(ctx)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return callback(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/commands/scheduler.py", line 178, in start_worker
    start_worker(queue, quiet = quiet)
      File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/background_jobs.py", line 155, in start_worker
return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    Worker(queues, name=get_worker_name(queue)).work(logging_level = logging_level)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 626, in work
        return self._parser.can_read(timeout)return ctx.invoke(self.callback, **ctx.params)

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    self.register_death()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 326, in register_death
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    return callback(*args, **kwargs)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/commands/scheduler.py", line 178, in start_worker
    p.execute()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 4012, in execute
        start_worker(queue, quiet = quiet)raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/apps/frappe/frappe/utils/background_jobs.py", line 155, in start_worker
redis.exceptions.ConnectionError: Connection closed by server.
    Worker(queues, name=get_worker_name(queue)).work(logging_level = logging_level)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 626, in work
    self.register_death()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/rq/worker.py", line 326, in register_death
    p.execute()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 4019, in execute
    conn = self.connection_pool.get_connection('MULTI',
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 563, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost:11000. Connection refused.
    return execute(conn, stack, raise_on_error)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3894, in _execute_transaction
    self.parse_response(connection, '_')
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3977, in parse_response
    result = Redis.parse_response(
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 915, in parse_response
    response = connection.read_response()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 739, in read_response
    response = self._parser.read_response()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 470, in read_response
    self.read_from_socket()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 452, in read_from_socket
    raise ConnectionError("Error while reading from socket: %s" %
redis.exceptions.ConnectionError: Error while reading from socket: (104, 'Connection reset by peer')
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
Exception in thread Exception in thread Exception in thread Thread-1Thread-1:
:
Thread-1Traceback (most recent call last):
Traceback (most recent call last):
:
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 973, in _bootstrap_inner
    self.run()
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
        self.run()
self.run()  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3718, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3617, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/client.py", line 3503, in parse_response
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    if not block and not conn.can_read(timeout=timeout):
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 734, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    return self._parser.can_read(timeout)
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 416, in can_read
    return self.read_from_socket(timeout=timeout,
  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
        return self.read_from_socket(timeout=timeout,raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)

  File "/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/redis/connection.py", line 429, in read_from_socket
redis.exceptions.ConnectionError: Connection closed by server.
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR)
redis.exceptions.ConnectionError: Connection closed by server.
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  from imp import reload
/home/ahmed/Desktop/ERPNEXT/frappe-bench/env/lib/python3.9/site-packages/bleach_whitelist/bleach_whitelist.py:3: DeprecationWarning: bleach-whitelist has been renamed bleach-allowlist. It will not receive further updates under the old name. See https://pypi.org/project/bleach-allowlist
  warnings.warn(
